walnut/converter/converter.go

```diff
@@ -0,0 +1,156 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package converter
+
+import (
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+	"os"
+
+	"github.com/AliceO2Group/Control/common"
+	"github.com/AliceO2Group/Control/common/controlmode"
+	"github.com/AliceO2Group/Control/common/gera"
+	"github.com/AliceO2Group/Control/core/task"
+	"github.com/AliceO2Group/Control/core/task/channel"
+)
+
+// return pointer to float64
+func createFloat(x float64) *float64 {
+	return &x
+}
+
+// return pointer to bool
+func createBool(x bool) *bool {
+	return &x
+}
+
+// return pointer to string
+func createString(x string) *string {
+	return &x
+}
+
+// ExtractTaskClasses takes in a DPL Dump string and extracts
+// an array of Tasks
+func ExtractTaskClasses(DPL Dump) (tasks []*task.Class, err error) {
+
+	for index := range DPL.Workflows {
+		var channelName string
+		if index+1 == len(DPL.Workflows) {
+			channelName = DPL.Workflows[index].Name
+		} else {
+			channelName = "from_" + DPL.Workflows[index].Name + "_to_" + DPL.Workflows[index+1].Name
+		}
+
+		taskName := DPL.Workflows[index].Name
+		defaultBindChannel := channel.Inbound{
+			Channel: channel.Channel{
+				Name:        channelName,
+				Type:        channel.ChannelType("push"), // defaulting to push
+				SndBufSize:  1000,
+				RcvBufSize:  1000,
+				RateLogging: 60,
+				Transport:   channel.TransportType("shmem"),
+			},
+			Addressing: "ipc",
+		}
+
+		// Not required for Task Templates
+		/*
+			defaultConnectChannel := channel.Outbound{
+				Channel: channel.Channel{
+					Name:        workflowName,
+					Type:        channel.ChannelType(""),
+					SndBufSize:  1000,
+					RcvBufSize:  1000,
+					RateLogging: 60,
+					Transport:   channel.TransportType("shmem"),
+				},
+				// Target: "", No default value
+			}
+		*/
+
+		task := task.Class{
+			Identifier: task.TaskClassIdentifier{
+				Name: taskName,
+			},
+			Defaults: gera.MakeStringMapWithMap(map[string]string{
+				"user": "flp",
+			}),
+			Control: struct {
+				Mode controlmode.ControlMode "yaml:\"mode\""
+			}{Mode: controlmode.FAIRMQ},
+			Command: &common.CommandInfo{
+				// Env: []string, -> Default to empty array
+				Shell:     createBool(true),
+				Value:     &DPL.Metadatas[index+1].Executable,
```

Comment:
"Data" and its compounds are singular uncountable. Please use "Metadata" instead of "Metadatas".

Reply:
Ah, it was simply to differentiate the array of `metadata` from each item. I'll make the change.

Reply:
The single item can be `entry`, `metadataEntry`, `metadataItem`, etc.